staging/src/k8s.io/pod-security-admission/policy/registry.go

```diff
@@ -0,0 +1,185 @@
+		return nil, nil
+	} else if !level.Valid() {
+		return nil, fmt.Errorf("invalid level %s", level)
+	}
+	var checks []Check
+
+	// baseline checks are included for both the baseline & restricted levels
+	for _, versionedChecks := range r.baselineChecks {
+		for _, check := range versionedChecks {
+			if !version.Older(&check.firstVersion) {
+				checks = append(checks, check)
+				break
+			}
+		}
+	}
+	if level == api.LevelBaseline {
+		return checks, nil
+	}
+
+	for _, versionedChecks := range r.restrictedChecks {
+		for _, check := range versionedChecks {
+			if !version.Older(&check.firstVersion) {
+				checks = append(checks, check)
+				break
+			}
+		}
+	}
+	return checks, nil
+}
+
+// AddCheck registers a Check at the given level. The checks are represented as a map of version ->
+// Check, where the version represents the first version that the associated check should be used.
+// All checks must answer the same ID. If the id is already registered, an error is returned.
+// Checks can only be added for baseline and restricted levels.
+func (r *CheckRegistry) AddCheck(level api.Level, checks map[string]Check) error {
+	id := ""
+	versionedChecks := make([]versionedCheck, 0, len(checks))
+	for v, check := range checks {
+		if check.ID() == "" {
+			return fmt.Errorf("missing ID for check version %s", v)
+		}
+		if id == "" {
+			id = check.ID()
+		} else if id != check.ID() {
+			return fmt.Errorf("check ID mismatch: %s != %s (%s)", id, check.ID(), v)
+		}
+		version, err := api.VersionToEvaluate(v)
+		if err != nil {
+			return fmt.Errorf("failed to parse version %s: %w", v, err)
+		} else if version.Latest() {
+			return errors.New("cannot register add a check for the 'latest' version")
+		}
+		versionedChecks = append(versionedChecks, versionedCheck{version, check})
+	}
+
+	if _, ok := r.restrictedChecks[id]; ok {
+		return fmt.Errorf("check %s already registered as under restricted", id)
+	}
+	if _, ok := r.baselineChecks[id]; ok {
+		return fmt.Errorf("check %s already registered as under baseline", id)
+	}
+
+	sort.Slice(versionedChecks, func(i, j int) bool {
+		// Newest checks first
+		return !versionedChecks[i].firstVersion.Older(&versionedChecks[j].firstVersion)
+	})
+
+	switch level {
+	case api.LevelRestricted:
+		r.restrictedChecks[id] = versionedChecks
+	case api.LevelBaseline:
+		r.baselineChecks[id] = versionedChecks
+	case api.LevelPrivileged:
+		return errors.New("cannot register checks for the privileged level")
+	default:
+		return fmt.Errorf("unknown level: %s", level)
+	}
+
+	return nil
+}
+
+func CheckForIDAndVersion(id string, version api.Version) (Check, error) {
+	return defaultRegistry.CheckForIDAndVersion(id, version)
+}
+func ChecksForLevelAndVersion(level api.Level, version api.Version) ([]Check, error) {
+	return defaultRegistry.ChecksForLevelAndVersion(level, version)
+}
+func DefaultRegistry() Registry {
+	return defaultRegistry
+}
+
+var defaultRegistry = NewCheckRegistry()
```

Comment:
I prefer not having globals.  It seems like we could handle this similar to schemes where an instance is instantiated.

Reply:
will rework to register a list of default checks and make a InstallChecks(registry) method to install those checks in a particular registry instance

Reply:
done